remove dependency on python cookbook

shawnashlee opened the pull request:
the python cookbook has been deprecated since 2014, lets
remove our usage of it. removing it will also allow the this
cookbook to be used with chef 13+

shawnashlee committed:
remove dependency on python cookbook

the python cookbook has been deprecated since 2014, lets
remove our usage of it. removing it will also allow the this
cookbook to be used with chef 13+

shawnashlee committed:
version bump to 0.0.6

zzantozz commented:
```
==> ele16:   ================================================================================
==> ele16:   Recipe Compile Error in /var/chef/cache/cookbooks/mimic/recipes/default.rb
==> ele16:   ================================================================================
==> ele16:   
==> ele16:   
==> ele16: SyntaxError
==> ele16:   -----------
==> ele16:   /var/chef/cache/cookbooks/mimic/recipes/default.rb:66: syntax error, unexpected end-of-input, expecting keyword_end
```

shawnashlee committed:
add missing `end` to bash block

missed this as Im used to the auto syntax checks in the chef
repo, we should add the same checks to this repo.

shawnashlee commented:
> ```
> ==> ele16:   /var/chef/cache/cookbooks/mimic/recipes/default.rb:66: syntax error, unexpected end-of-input, expecting keyword_end
> ```

fixed in 09ffa17

zzantozz commented:
Still

```
==> ele16:   ================================================================================
==> ele16:   
==> ele16: Recipe Compile Error in /var/chef/cache/cookbooks/mimic/recipes/default.rb
==> ele16:   
==> ele16: ================================================================================
==> ele16:   
==> ele16: 
==> ele16:   
==> ele16: SyntaxError
==> ele16:   -----------
==> ele16:   
==> ele16: /var/chef/cache/cookbooks/mimic/recipes/default.rb:67: syntax error, unexpected end-of-input, expecting keyword_end
==> ele16:   
```

shawnashlee committed:
add missing `end` to bash block

copy paste fail, this adds a missing `end` to a bash block

shawnashlee commented:
this was fixed last week, still looking for a +1
```
vagrant@ubuntu-xenial:/vagrant$ sudo chef-client -o 'recipe[runit],recipe[mimic]'
Starting Chef Client, version 12.22.5
resolving cookbooks for run list: ["runit", "mimic"]
Synchronizing Cookbooks:
  - runit (4.3.1)
  - mimic (0.0.6)
  - packagecloud (1.0.1)
  - yum-epel (3.3.0)
Installing Cookbook Gems:
Compiling Cookbooks...
Converging 10 resources
Recipe: runit::default
  * apt_package[runit] action install
  Recipe: <Dynamically Defined Resource>
    * cookbook_file[/opt/chef/cache/preseed/runit/runit-2.1.2-3ubuntu1.seed] action create
      - create new file /opt/chef/cache/preseed/runit/runit-2.1.2-3ubuntu1.seed
      - update content in file /opt/chef/cache/preseed/runit/runit-2.1.2-3ubuntu1.seed from none to 9c6758
      --- /opt/chef/cache/preseed/runit/runit-2.1.2-3ubuntu1.seed       2023-08-01 22:55:29.278018000 +0000
      +++ /opt/chef/cache/preseed/runit/.chef-runit-220230801-4473-ls4t24.1.2-3ubuntu1.seed     2023-08-01 22:55:29.278018000 +0000                                                                                                 
      @@ -1 +1,2 @@
      +runit   runit/signalinit        boolean true
    - preseed package runit
    - install version 2.1.2-3ubuntu1 of package runit
Recipe: runit::default
  * service[runit] action start (up to date)
  * service[runit] action enable (up to date)
Recipe: mimic::default
  * apt_package[python-virtualenv] action install
    - install version 15.0.1+ds-3ubuntu1.1 of package python-virtualenv
  * apt_package[libffi6] action install (up to date)
  * apt_package[libffi-dev] action install
    - install version 3.2.1-4 of package libffi-dev
  * bash[delete virtualenv /srv/mimic] action run (skipped due to only_if)
  * bash[create virtualenv /srv/mimic] action run
    [execute] New python executable in /srv/mimic/bin/python2
              Also creating executable in /srv/mimic/bin/python
              Installing setuptools < 45, pkg_resources, pip < 21, wheel...done.
              Running virtualenv with interpreter /usr/bin/python2
    - execute "bash"  "/tmp/chef-script20230801-4473-18vdlfv"
  * bash[install mimic] action run
    [execute] DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
              WARNING: The directory '/home/vagrant/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
              Collecting mimic
                Downloading mimic-2.2.0-py2-none-any.whl (283 kB)
              Collecting xmltodict>=0.9.1
                Downloading xmltodict-0.12.0-py2.py3-none-any.whl (9.2 kB)
              Collecting testtools<1.8.0,>=1.7.1
                Downloading testtools-1.7.1-py2.py3-none-any.whl (159 kB)
              Collecting attrs>=15.1.0
                Downloading attrs-21.4.0-py2.py3-none-any.whl (60 kB)
              Collecting iso8601>=0.1.10
                Downloading iso8601-0.1.16-py2.py3-none-any.whl (10 kB)
              Collecting treq>=15.1.0
                Downloading treq-21.1.0-py2.py3-none-any.whl (64 kB)
              Collecting twisted>=15.5.0
                Downloading Twisted-20.3.0-cp27-cp27mu-manylinux1_x86_64.whl (3.2 MB)
              Collecting six>=1.6.0
                Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
              Collecting klein>=15.3.1
                Downloading klein-20.6.0-py2.py3-none-any.whl (118 kB)
              Collecting traceback2
                Downloading traceback2-1.4.0-py2.py3-none-any.whl (16 kB)
              Collecting python-mimeparse
                Downloading python_mimeparse-1.6.0-py2.py3-none-any.whl (6.1 kB)
              Collecting extras
                Downloading extras-1.0.0-py2.py3-none-any.whl (7.3 kB)
              Collecting unittest2>=1.0.0
                Downloading unittest2-1.1.0-py2.py3-none-any.whl (96 kB)
              Collecting requests>=2.1.0
                Downloading requests-2.27.1-py2.py3-none-any.whl (63 kB)
              Collecting hyperlink>=21.0.0
                Downloading hyperlink-21.0.0-py2.py3-none-any.whl (74 kB)
              Collecting incremental
                Downloading incremental-22.10.0-py2.py3-none-any.whl (16 kB)
              Collecting PyHamcrest!=1.10.0,>=1.9.0
                Downloading PyHamcrest-1.10.1.tar.gz (43 kB)
              Collecting constantly>=15.1
                Downloading constantly-15.1.0-py2.py3-none-any.whl (7.9 kB)
              Collecting Automat>=0.3.0
                Downloading Automat-22.10.0-py2.py3-none-any.whl (26 kB)
              Collecting zope.interface>=4.4.2
                Downloading zope.interface-5.5.2.tar.gz (300 kB)
              Collecting Werkzeug
                Downloading Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
              Collecting enum34; python_version < "3.4"
                Downloading enum34-1.1.10-py2-none-any.whl (11 kB)
              Collecting Tubes
                Downloading Tubes-0.2.1-py2.py3-none-any.whl (59 kB)
              Collecting typing; python_version < "3.5"
                Downloading typing-3.10.0.0-py2-none-any.whl (26 kB)
              Collecting linecache2
                Downloading linecache2-1.0.0-py2.py3-none-any.whl (12 kB)
              Requirement already satisfied: argparse in /usr/lib/python2.7 (from unittest2>=1.0.0->testtools<1.8.0,>=1.7.1->mimic) (1.2.1)
              Collecting certifi>=2017.4.17
                Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
              Collecting urllib3<1.27,>=1.21.1
                Downloading urllib3-1.26.16-py2.py3-none-any.whl (143 kB)
              Collecting idna<3,>=2.5; python_version < "3"
                Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
              Collecting chardet<5,>=3.0.2; python_version < "3"
                Downloading chardet-4.0.0-py2.py3-none-any.whl (178 kB)
              Requirement already satisfied: setuptools in /srv/mimic/lib/python2.7/site-packages (from zope.interface>=4.4.2->twisted>=15.5.0->mimic) (44.1.1)
              Building wheels for collected packages: PyHamcrest, zope.interface
                Building wheel for PyHamcrest (setup.py): started
                Building wheel for PyHamcrest (setup.py): finished with status 'done'
                Created wheel for PyHamcrest: filename=PyHamcrest-1.10.1-py2-none-any.whl size=48898 sha256=5febe2bb80116dcbbe087237d8250b7ae7dd93539fbec65a244ab8dad80287c2
                Stored in directory: /tmp/pip-ephem-wheel-cache-84n6Jf/wheels/f5/8c/e2/f0cea19d340270166bbfd4a2e9d8b8c132e26ef7e1376a0890
                Building wheel for zope.interface (setup.py): started
                Building wheel for zope.interface (setup.py): finished with status 'done'
                Created wheel for zope.interface: filename=zope.interface-5.5.2-cp27-cp27mu-linux_x86_64.whl size=198975 sha256=45da7fa42cf94cdb524fd9e90e437ca3a4ec39f2b630ad04ebb22227ca36cc60
                Stored in directory: /tmp/pip-ephem-wheel-cache-84n6Jf/wheels/8f/60/b0/736e361327f969f38ad9ee2e24132600a82fac63ab43b7082e
              Successfully built PyHamcrest zope.interface
              Installing collected packages: xmltodict, linecache2, traceback2, python-mimeparse, extras, six, unittest2, testtools, attrs, iso8601, certifi, urllib3, idna, chardet, requests, typing, hyperlink, incremental, PyHamcrest, constantly, Automat, zope.interface, twisted, treq, Werkzeug, enum34, Tubes, klein, mimic
              Successfully installed Automat-22.10.0 PyHamcrest-1.10.1 Tubes-0.2.1 Werkzeug-1.0.1 attrs-21.4.0 certifi-2021.10.8 chardet-4.0.0 constantly-15.1.0 enum34-1.1.10 extras-1.0.0 hyperlink-21.0.0 idna-2.10 incremental-22.10.0 iso8601-0.1.16 klein-20.6.0 linecache2-1.0.0 mimic-2.2.0 python-mimeparse-1.6.0 requests-2.27.1 six-1.16.0 testtools-1.7.1 traceback2-1.4.0 treq-21.1.0 twisted-20.3.0 typing-3.10.0.0 unittest2-1.1.0 urllib3-1.26.16 xmltodict-0.12.0 zope.interface-5.5.2
    - execute "bash"  "/tmp/chef-script20230801-4473-bkbvh0"
Recipe: <Dynamically Defined Resource>
  * service[mimic] action nothing (skipped due to action :nothing)
Recipe: mimic::default
  * runit_service[mimic] action enable
    * ruby_block[restart_service] action nothing (skipped due to action :nothing)
    * ruby_block[restart_log_service] action nothing (skipped due to action :nothing)
    * directory[/etc/sv/mimic] action create
      - create new directory /etc/sv/mimic
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * template[/etc/sv/mimic/run] action create
      - create new file /etc/sv/mimic/run
      - update content in file /etc/sv/mimic/run from none to bdff91
      --- /etc/sv/mimic/run     2023-08-01 22:56:04.983862000 +0000
      +++ /etc/sv/mimic/.chef-run20230801-4473-tqh804   2023-08-01 22:56:04.983862000 +0000
      @@ -1 +1,11 @@
      +#!/bin/bash
      +
      +NAME=mimic
      +DIR=/opt/mimic
      +TWISTD=/srv/mimic/bin/twistd
      +OWNER=daemon
      +
      +logger -p daemon.info -t runit "${NAME} service starting"
      +export PYTHONPATH=${DIR}:${PYTHONPATH}
      +exec chpst -u ${OWNER} ${TWISTD} -n ${NAME} --realtime --verbose
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * directory[/etc/sv/mimic/log] action create
      - create new directory /etc/sv/mimic/log
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * directory[/etc/sv/mimic/log/main] action create
      - create new directory /etc/sv/mimic/log/main
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * directory[/var/log/mimic] action create
      - create new directory /var/log/mimic
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * template[/etc/sv/mimic/log/config] action create
      - create new file /etc/sv/mimic/log/config
      - update content in file /etc/sv/mimic/log/config from none to e3b0c4
      (no diff)
      - change mode from '' to '0644'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * link[/var/log/mimic/config] action create
      - create symlink at /var/log/mimic/config to /etc/sv/mimic/log/config
    * template[/etc/sv/mimic/log/run] action create
      - create new file /etc/sv/mimic/log/run
      - update content in file /etc/sv/mimic/log/run from none to 0f0195
      --- /etc/sv/mimic/log/run 2023-08-01 22:56:05.039889999 +0000
      +++ /etc/sv/mimic/log/.chef-run20230801-4473-d2zivs       2023-08-01 22:56:05.039889999 +0000
      @@ -1 +1,3 @@
      +#!/bin/sh
      +exec svlogd -tt /var/log/mimic
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * directory[/etc/sv/mimic/env] action create
      - create new directory /etc/sv/mimic/env
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * ruby_block[Delete unmanaged env files for mimic service] action run (skipped due to only_if)
    * template[/etc/sv/mimic/check] action create (skipped due to only_if)
    * template[/etc/sv/mimic/finish] action create
      - create new file /etc/sv/mimic/finish
      - update content in file /etc/sv/mimic/finish from none to a42937
      --- /etc/sv/mimic/finish  2023-08-01 22:56:05.055897999 +0000
      +++ /etc/sv/mimic/.chef-finish20230801-4473-1qz3chl       2023-08-01 22:56:05.055897999 +0000
      @@ -1 +1,10 @@
      +#!/bin/bash
      +
      +if [ $1 -ne 0 ]; then
      +  PRIORITY=daemon.error
      +else
      +  PRIORITY=daemon.info
      +fi
      +
      +logger -p ${PRIORITY} -t runit "mimic exited with status $1"
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * directory[/etc/sv/mimic/control] action create
      - create new directory /etc/sv/mimic/control
      - change mode from '' to '0755'
      - change owner from '' to 'daemon'
      - change group from '' to 'daemon'
    * ruby_block[unlink /etc/init.d/mimic] action run (skipped due to only_if)
    * template[/etc/init.d/mimic] action create
      - create new file /etc/init.d/mimic
      - update content in file /etc/init.d/mimic from none to d92750
      --- /etc/init.d/mimic     2023-08-01 22:56:05.071906000 +0000
      +++ /etc/init.d/.chef-mimic20230801-4473-1ihfggb  2023-08-01 22:56:05.071906000 +0000
      @@ -1 +1,93 @@
      +#!/bin/sh
      +### BEGIN INIT INFO
      +# Provides:          mimic
      +# Required-Start:
      +# Required-Stop:
      +# Default-Start:
      +# Default-Stop:
      +# Short-Description: initscript for runit-managed mimic service
      +### END INIT INFO
      +
      +# Author: Chef Software, Inc. <cookbooks@chef.io>
      +
      +PATH=/sbin:/usr/sbin:/bin:/usr/bin
      +DESC="runit-managed mimic"
      +NAME=mimic
      +RUNIT=/usr/bin/sv
      +RUNIT_ARGS=""
      +SCRIPTNAME=/etc/init.d/$NAME
      +
      +# Exit if runit is not installed
      +[ -x $RUNIT ] || exit 0
      +
      +# Load the VERBOSE setting and other rcS variables
      +. /lib/init/vars.sh
      +
      +# Define LSB log_* functions.
      +# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
      +. /lib/lsb/init-functions
      +
      +
      +case "$1" in
      +  start)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Starting $DESC " "$NAME"
      +        $RUNIT $RUNIT_ARGS start $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  stop)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Stopping $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS stop $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  status)
      +        $RUNIT $RUNIT_ARGS status $NAME && exit 0 || exit $?
      +        ;;
      +  reload)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Reloading $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS reload $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  force-reload)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Force reloading $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS force-reload $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  force-stop)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Force stopping $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS force-stop $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  force-restart)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Force restarting $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS force-restart $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  force-shutdown)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Force shutdowning $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS force-shutdown $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  restart)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Restarting $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS restart $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  shutdown)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Shutdowning $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS shutdown $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  try-restart)
      +        [ "$VERBOSE" != no ] && log_daemon_msg "Try restarting $DESC" "$NAME"
      +        $RUNIT $RUNIT_ARGS try-restart $NAME
      +        [ "$VERBOSE" != no ] && log_end_msg $?
      +        ;;
      +  *)
      +        echo "Usage: $SCRIPTNAME {start|stop|status|reload|force-reload|force-restart|force-shutdown|force-stop|restart|shutdown|try-restart}" >&2                                                                           
      +        exit 3
      +        ;;
      +esac
      +
      +:
      +
      - change mode from '' to '0755'
      - change owner from '' to 'root'
      - change group from '' to 'root'
    * file[/etc/sv/mimic/down] action nothing (skipped due to action :nothing)
    * ruby_block[restart_service] action run
      * ruby_block[restart_service] action nothing (skipped due to action :nothing)
      * ruby_block[restart_log_service] action nothing (skipped due to action :nothing)
      * directory[/etc/sv/mimic] action create (up to date)
      * template[/etc/sv/mimic/run] action create (up to date)
      * directory[/etc/sv/mimic/log] action create (up to date)
      * directory[/etc/sv/mimic/log/main] action create (up to date)
      * directory[/var/log/mimic] action create (up to date)
      * template[/etc/sv/mimic/log/config] action create (up to date)
      * link[/var/log/mimic/config] action create (up to date)
      * template[/etc/sv/mimic/log/run] action create (up to date)
      * directory[/etc/sv/mimic/env] action create (up to date)
      * ruby_block[Delete unmanaged env files for mimic service] action run (skipped due to only_if)
      * template[/etc/sv/mimic/check] action create (skipped due to only_if)
      * template[/etc/sv/mimic/finish] action create (up to date)
      * directory[/etc/sv/mimic/control] action create (up to date)
      * ruby_block[unlink /etc/init.d/mimic] action run (skipped due to only_if)
      * template[/etc/init.d/mimic] action create (up to date)
      * file[/etc/sv/mimic/down] action nothing (skipped due to action :nothing)
      * directory[/etc/service] action create (up to date)
      * link[/etc/service/mimic] action create
        - create symlink at /etc/service/mimic to /etc/sv/mimic
      * ruby_block[wait for mimic service socket] action run
        - execute the ruby block wait for mimic service socket
      - execute the ruby block restart_service
    * ruby_block[restart_log_service] action run
      * ruby_block[restart_service] action nothing (skipped due to action :nothing)
      * ruby_block[restart_log_service] action nothing (skipped due to action :nothing)
      * directory[/etc/sv/mimic] action create (up to date)
      * template[/etc/sv/mimic/run] action create (up to date)
      * directory[/etc/sv/mimic/log] action create (up to date)
      * directory[/etc/sv/mimic/log/main] action create (up to date)
      * directory[/var/log/mimic] action create (up to date)
      * template[/etc/sv/mimic/log/config] action create (up to date)
      * link[/var/log/mimic/config] action create (up to date)
      * template[/etc/sv/mimic/log/run] action create (up to date)
      * directory[/etc/sv/mimic/env] action create (up to date)
      * ruby_block[Delete unmanaged env files for mimic service] action run (skipped due to only_if)
      * template[/etc/sv/mimic/check] action create (skipped due to only_if)
      * template[/etc/sv/mimic/finish] action create (up to date)
      * directory[/etc/sv/mimic/control] action create (up to date)
      * ruby_block[unlink /etc/init.d/mimic] action run (skipped due to only_if)
      * template[/etc/init.d/mimic] action create (up to date)
      * file[/etc/sv/mimic/down] action nothing (skipped due to action :nothing)
      * directory[/etc/service] action create (up to date)
      * link[/etc/service/mimic] action create (up to date)
      * ruby_block[wait for mimic service socket] action run
        - execute the ruby block wait for mimic service socket
      - execute the ruby block restart_log_service
    * directory[/etc/service] action create (up to date)
    * link[/etc/service/mimic] action create (up to date)
    * ruby_block[wait for mimic service socket] action run
      - execute the ruby block wait for mimic service socket
  
  * runit_service[mimic] action start (up to date)

Running handlers:
Running handlers complete
Chef Client finished, 25/78 resources updated in 41 seconds
```

zzantozz commented:
Cool, I just hadn't taken time to test it out yet.
